nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestAttributesToJSON.java

```diff
@@ -560,8 +555,34 @@ public void testAttributeWithNestedJsonOutputAsStringInAttribute(String nestedJs
 
         String json = testRunner.getFlowFilesForRelationship(AttributesToJSON.REL_SUCCESS)
                 .get(0).getAttribute(AttributesToJSON.JSON_ATTRIBUTE_NAME);
-        ObjectMapper mapper = new ObjectMapper();
-        Map<String, Object> val = mapper.readValue(json, Map.class);
+        Map<String, Object> val = MAPPER.readValue(json, Map.class);
         assertInstanceOf(String.class, val.get(TEST_ATTRIBUTE_KEY));
     }
+
+    @Test
+    public void testAttributeWithJsonBetweenLeadingAndTrailingSpaces() throws JsonProcessingException {
+        final String jsonBetweenLeadingAndTrailingSpaces = "\n{\n" +
+                "    \"fruit\": \"Apple\",\n" +
+                "    \"size\": \"Large\",\n" +
+                "    \"color\": \"Red\"\n" +
+                "}\n";
+        assertDoesNotThrow((ThrowingSupplier<Object>) () -> MAPPER.readValue(jsonBetweenLeadingAndTrailingSpaces, Map.class));
```

Comment:
It should not be necessary to cast the supplier:
```suggestion
        assertDoesNotThrow(() -> MAPPER.readValue(jsonBetweenLeadingAndTrailingSpaces, Map.class));
```